modules/nf-core/bbmap/bbmerge/main.nf

```diff
@@ -0,0 +1,59 @@
+process BBMAP_BBMERGE {
+    tag "$meta.id"
+    label 'process_single'
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bbmap:39.06--h92535d8_0':
+        'biocontainers/bbmap:39.06--h92535d8_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*_merged.fastq"), emit: merged
+    tuple val(meta), path("*_unmerged.fastq"), emit: unmerged
+    tuple val(meta), path("*_ihist.txt"), emit: ihist
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    maxmem=\$(echo \"$task.memory\"| sed 's/ GB/g/g')
+    bbmerge.sh \\
+        -Xmx\$maxmem \\
+        in1=${reads[0]} \\
+        in2=${reads[1]} \\
+        out=${prefix}_merged.fastq \\
+        outu1=${prefix}_1_unmerged.fastq \\
+        outu2=${prefix}_2_unmerged.fastq \\
```

Comment:
I believe this command is not going to work for interleaved reads from a single file. When the user provides a single file, there will be an issue trying to access reads[1]. Maybe you could deal with it in a similar way as they do here and provide an input that indicates the use of an interleaved file: https://github.com/nf-core/modules/blob/master/modules/nf-core/samtools/fastq/main.nf Then it would also be possible to adapt the output to the input type.